triton-docker

```diff
@@ -71,6 +59,58 @@ function install () {
     echo
 }
 
+# Emit the current Triton profile and time
+function currentProfile () {
+    if [ -n "$TRITON_PROFILE" ]
+    then
+        local profile_source='via env var; use `eval "$(triton env <profile name>)"` to change'
+    else
+        local profile_source='default; use `triton profile set <profile name>` to change'
+    fi
+
+    echo "Executing in '$(triton profile get | awk 'NR == 1 {print $2}')' (${profile_source})"
+}
+
+# Extra env vars
+function extraEnvVars () {
+
+    # If the CNS env vars are already set, don't bother continuing
+    if [ -n "$TRITON_CNS_SUFFIX_PUBLIC" ] && [ -n "$TRITON_CNS_SUFFIX_PRIVATE" ]
+    then
+        return
+    fi
+
+    # Get and the user's account information for parsing later
+    local triton_account="$(triton account get)"
+
+    # Check if CNS is enabled, require it
+    local triton_cns_enabled="$(echo "$triton_account" | awk -F": " '/cns/{print $2}')"
+    if [ ! "true" == "$triton_cns_enabled" ]; then
+        echo
+        tput rev  # reverse
+        tput bold # bold
+        echo 'Error! Triton CNS is required and not enabled.'
+        tput sgr0 # clear
+        echo
+        echo 'Please run:'
+        echo '  triton account update triton_cns_enabled=true'
+        echo
+        exit 1
+    fi
+
+    # Get the user's UUID
+    local triton_account_uuid="$(echo "$triton_account" | awk -F": " '/id:/{print $2}')"
+
+    # Get current data center name
+    # Note: this makes assumptions that work in our public cloud, but might not work elsewhere
+    # Further note: set TRITON_CNS_SUFFIX_PUBLIC and TRITON_CNS_SUFFIX_PRIVATE to work around this
+    local triton_dc="$(triton profile get | awk -F"/" '/url:/{print $3}' | awk -F'.' '{print $1}')"
```

Comment:
This is pretty brittle, e.g. for my `staging1` profile:

```
$ triton profile get staging1 | awk -F"/" '/url:/{print $3}' | awk -F'.' '{print $1}'
cloudapi
```

I think the best current way to reliably get the DC name is from ListDatacenters:

```
$ triton datacenters
NAME       URL
eu-ams-1   https://eu-ams-1.api.joyentcloud.com
us-east-1  https://us-east-1.api.joyentcloud.com
us-east-2  https://us-east-2.api.joyentcloud.com
us-east-3  https://us-east-3.api.joyentcloud.com
us-sw-1    https://us-sw-1.api.joyentcloud.com
us-west-1  https://us-west-1.api.joyentcloud.com
```

... and pick out the one that matches your profile's `url`.  If you have a URL that resolves to the same, but isn't one of the ones in there, then you are screwed.  I.e. perhaps it would be good to have cloudapi do better and identify which one is the current one.

From discussion with @tgross yesterday, we were talking about having `triton` cache the account *UUID* at profile creation time. Perhaps it could get and cache the DC name as well (assuming that isn't going to change ever).